tests/test_problems.py

```diff
@@ -796,6 +839,38 @@ def test_assertion_fn():
 
         self.assertRaises(Exception, test_assertion_fn)
 
+    def test_problem_instance(self):
+        """
+        Test problem instance method of the QUBO class.
+        We create a random instance of all the different problems, we generate the QUBO problem out of it and then we check if the problem instance attribute is correct, by comparing the keys of the problem instance with the expected keys.
+        """
+
+        all = {
+            "tsp":TSP.random_instance(n_cities=3),
+            "number_partition":NumberPartition.random_instance(n_numbers=3),
+            "maximum_cut":MaximumCut.random_instance(n_nodes=3, edge_probability=0.5),
```

Comment:
can you put all both `n` and the `edge_prob` to take random values? It may be useful in the future, as they may give some edge cases :)

Reply:
done